src/libambit/device_driver_ambit3.c

```diff
@@ -371,6 +588,14 @@ static int parse_log_header(const uint8_t *data, ambit3_log_header_t *log_header
         free(log_header->header.activity_name);
     }
     log_header->header.activity_name = utf8memconv((const char*)(data + offset), 16, "ISO-8859-15");
+    switch (log_type) {
+      case 0x7e: /* gen1 fw */
+        break;
+      case 0x7a: /* gen2 fw */
+      case 0x8a: /* gen2 fw */
+        offset += (strlen(log_header->header.activity_name)+1);
```

Comment:
That sounds like a possibility indeed. Should probably strlen() the input data, rather than the UTF-8 converted one, and use that to offset the offset.